### PR TITLE
chore: Add logs and cluster objects collecting for tests

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Adds scripts that collects all pods logs in Che namespaces as well as most common objects yamls for integration tests.

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/20504

### How to test this PR?
Look at attached artifacts in GH actions for integration tests.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [Custom resource definition file is up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#updating-custom-resource-definition-file)
- [ ] [OLM bundles are up to date](https://github.com/eclipse-che/che-operator/blob/main/README.md#update-olm-bundle)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .github/bin/common.sh [hunk 1]
```diff
@@ -170,12 +170,48 @@ collectLogs() {
 
   set +e
   chectl server:logs --chenamespace=${NAMESPACE} --directory=${ARTIFACTS_DIR}
+  collectK8sResourcesForNamespace $NAMESPACE
+  collectPodsLogsForNamespace $NAMESPACE
```
Comment (tolusha) [inline .github/bin/common.sh:174]: We have to dump all namespaces

File: .github/bin/common.sh [hunk 2]
```diff
@@ -170,12 +170,48 @@ collectLogs() {
...
 }
 
+collectK8sResourcesForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
+
+  declare -a KINDS=("pods" "deployments" "services"
+                    "configmaps" "secrets"
+                    "serviceaccounts" "roles" "rolebindings"
+                    "checlusters" "checlusterbackups" "checlusterrestores"
```
Comment (tolusha) [inline .github/bin/common.sh:189]: Regarding CR..
Let's iterate over CRD and dump corresponding CR

File: .github/bin/common.sh (hunk 1 above)
Comment (mmorhun) [reply .github/bin/common.sh]: ok

File: .github/bin/common.sh [hunk 3]
```diff
@@ -170,12 +170,74 @@ collectLogs() {
 
   set +e
   chectl server:logs --chenamespace=${NAMESPACE} --directory=${ARTIFACTS_DIR}
+
+  collectClusterData
+
   collectDevworkspaceOperatorLogs
```
Comment (tolusha) [inline .github/bin/common.sh:176]: We don't need this anymore

File: .github/bin/common.sh [hunk 4]
```diff
@@ -170,12 +170,74 @@ collectLogs() {
...
   oc get events -n ${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE}.txt
   oc get events -n ${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE}.txt
```
Comment (tolusha) [inline .github/bin/common.sh:178]: We can move events collecting to new functions as well

File: .github/bin/common.sh (hunk 3 above)
Comment (mmorhun) [reply .github/bin/common.sh:176]: I think we do ;)

File: .github/bin/common.sh [hunk 5]
```diff
@@ -169,31 +169,74 @@ collectLogs() {
   mkdir -p ${ARTIFACTS_DIR}
 
   set +e
-  chectl server:logs --chenamespace=${NAMESPACE} --directory=${ARTIFACTS_DIR}
-  collectDevworkspaceOperatorLogs
-  oc get events -n ${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE}.txt
-  oc get events -n ${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE}.txt
+  collectClusterData
```
[the remark this answers is not in the document]
Comment (tolusha) [reply .github/bin/common.sh:172]: 4. Is `pv` a cluster scope object?

[the remark this answers is not in the document]
Comment (mmorhun) [reply .github/bin/common.sh:171]: ok

File: .github/bin/common.sh [hunk 6]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
+      kubectl logs -n $namespace $pod -c $container > "${dir}/${pod}_${container}.log"
+    done
   done
 }
 
+collectEventsForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
+
+  dir="${ARTIFACTS_DIR}/cluster/namespaces/${namespace}"
```
Comment (tolusha) [inline .github/bin/common.sh:246]: `resources/namespaced` ?
pls move to a const to reuse.

File: .github/bin/common.sh [hunk 7]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
-# Graps Eclipse Che logs
 collectLogs() {
   mkdir -p ${ARTIFACTS_DIR}
 
   set +e
-  chectl server:logs --chenamespace=${NAMESPACE} --directory=${ARTIFACTS_DIR}
-  collectDevworkspaceOperatorLogs
-  oc get events -n ${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CONTROLLER_TEST_NAMESPACE}.txt
-  oc get events -n ${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE} > ${ARTIFACTS_DIR}/events-${DEVWORKSPACE_CHE_OPERATOR_TEST_NAMESPACE}.txt
+  collectClusterData
```
Comment (tolusha) [inline .github/bin/common.sh:171]: collect(Resources|Artificats) ?

File: .github/bin/common.sh [hunk 8]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
-collectDevworkspaceOperatorLogs() {
-  mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator
-
-  oc get events -n devworkspace-controller > ${ARTIFACTS_DIR}/events-devworkspace-controller.txt
+collectClusterData() {
+  allNamespaces=$(kubectl get namespaces -o custom-columns=":metadata.name")
+  for namespace in $allNamespaces ; do
+    collectK8sResourcesForNamespace $namespace
+    collectPodsLogsForNamespace $namespace
+    collectEventsForNamespace $namespace
```
Comment (tolusha) [inline .github/bin/common.sh:180]:
```suggestion
    collectNamespacedScopeResources $namespace
    collectNamespacedLogs $namespace
    collectNamespacedEvents $namespace
```

File: .github/bin/common.sh [hunk 9]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
+  done
+  collectClusterScopeK8sResources
```
Comment (tolusha) [inline .github/bin/common.sh:182]:
```suggestion
  collectClusterScopeResources
```

File: .github/bin/common.sh [hunk 10]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
+
+collectK8sResourcesForNamespace() {
+  namespace="$1"
+  if [[ -z $namespace ]]; then return; fi
+
+  STANDARD_KINDS=("pods" "jobs" "deployments"
+                  "services" "ingresses"
+                  "configmaps" "secrets"
+                  "serviceaccounts" "roles" "rolebindings"
+                  "pv" "pvc"
```
Comment (tolusha) [inline .github/bin/common.sh:193]: Move `pv` to cluster scope

File: .github/bin/common.sh [hunk 11]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
+                   )
+  for kind in "${KINDS[@]}" ; do
+    dir="${ARTIFACTS_DIR}/resources/cluster/${kind}"
+    mkdir -p $dir
 
-  # save the logs of all the containers in the DWO pod
-  for container in $(oc get pod -n devworkspace-controller ${CONTROLLER_POD_NAME} -o json | jq -r '.spec.containers[] | .name'); do
-    mkdir -p ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}
-    oc logs -n devworkspace-controller deployment/devworkspace-controller-manager -c ${container} > ${ARTIFACTS_DIR}/devworkspace-operator/${CONTROLLER_POD_NAME}/${container}.log
+    names=$(kubectl get -n $namespace $kind --no-headers=true -o custom-columns=":metadata.name")
```
Comment (AndrienkoAleksandr) [inline .github/bin/common.sh:218]: If that's a cluster scope objects, why do we need namespace argument?

File: .github/bin/common.sh [hunk 12]
```diff
@@ -164,39 +164,91 @@ installYq() {
...
+    for name in $names ; do
+      name=${name//[:<>|*?]/_}
+      kubectl get -n $namespace $kind $name -o yaml > "${dir}/${name}.yaml"
```
Comment (AndrienkoAleksandr) [inline .github/bin/common.sh:221]: If that's a cluster scope objects, why do we need namespace argument?

File: .github/bin/common.sh (hunk 10 above)
Comment (mmorhun) [reply .github/bin/common.sh:193]: ok

File: .github/bin/common.sh (hunk 6 above)
Comment (mmorhun) [reply .github/bin/common.sh:246]: ok

File: .github/bin/common.sh (hunk 7 above)
Comment (mmorhun) [reply .github/bin/common.sh:171]: `collectClusterResources`

File: .github/bin/common.sh (hunk 11 above)
Comment (mmorhun) [reply .github/bin/common.sh:206]: This function is designed for namespaced objects only. There is a separate one for luster scope objects.

**Review by tolusha — APPROVED** [review]

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**